### PR TITLE
Improvements

### PR DESCRIPTION
**Pull request by ProfliX**
Removed redundant call toString()
Optimized imports
Rearanged/refactored code (for better readability)
Val(same as var but with final) and var (lombok) usement for whole project (where is actually needed).
Unused code suppressed
Removed Map,Collection,List raw types (Transformed them to valid ones)
FileUtils > extractResource() > Asserted not null inputStream
DataStorage > getEnum > assertet object instanceof String;
Added JavaDocs to ConfigSettings and ReloadSettings
LightningSerializer > deserialize improvement
LightningException > Removed NotNull annotation from primitive type members.
Bumped new version > 3.2.4

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 1]
```diff
@@ -1,7 +1,5 @@
 ---
-name: Bug report
-about: Create a report to help us improve
-title: ''
+name: Bug report about: Create a report to help us improve title: ''
```
Comment (yoyosource) [inline .github/ISSUE_TEMPLATE/bug_report.md:2]: What happened here? This is unintended right?

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 2]
```diff
@@ -10,6 +8,7 @@ A clear and concise description of what the bug is.
 
 **To Reproduce**
 Steps to reproduce the behavior:
+
```
Comment (yoyosource) [inline .github/ISSUE_TEMPLATE/bug_report.md:11]: Why a new line?

File: .github/ISSUE_TEMPLATE/bug_report.md [hunk 3]
```diff
@@ -22,10 +21,11 @@ A clear and concise description of what you expected to happen.
 If applicable, add screenshots to help explain your problem.
 
 **Desktop (please complete the following information):**
- - OS: [e.g. Ubuntu]
- - Version [e.g. 18.04]
-**Java**
- - Version: [e.g. 1.8.0_291]
+
```
Comment (yoyosource) [inline .github/ISSUE_TEMPLATE/bug_report.md:24]: Why a new line? See above!

File: .github/ISSUE_TEMPLATE/feature_request.md [hunk 4]
```diff
@@ -1,7 +1,5 @@
 ---
-name: Feature request
-about: Suggest an idea for this project
-title: ''
+name: Feature request about: Suggest an idea for this project title: ''
```
Comment (yoyosource) [inline .github/ISSUE_TEMPLATE/feature_request.md:2]: What happened here? This is unintended right? See above!

File: src/main/java/de/leonhard/storage/internal/provider/LightningProviders.java [hunk 5]
```diff
@@ -14,54 +15,54 @@
...
 
-  @Setter
-  private MapProvider mapProvider;
-  @Setter
-  private YamlConfig yamlConfig;
-  @Setter
-  private InputStreamProvider inputStreamProvider;
-  @Setter
-  private ExceptionHandler exceptionHandler;
+    @Setter
```
Comment (yoyosource) [inline src/main/java/de/leonhard/storage/internal/provider/LightningProviders.java:18]: If this are the only fields of this class you can also put `@Setter` on the class itself. But pay attention to the class fields, when there are some without a setter.

File: src/main/java/de/leonhard/storage/internal/settings/DataType.java [hunk 6]
```diff
@@ -2,40 +2,45 @@
...
+        public @NotNull Map<String, Object> getMapImplementation() {
+            return mapProvider.getSortedMapImplementation();
+        }
+    },
+
+    // Unsorted data type
+    UNSORTED {
+        @Override
+        public @NotNull Map<String, Object> getMapImplementation() {
+            return mapProvider.getSortedMapImplementation();
```
Comment (yoyosource) [inline src/main/java/de/leonhard/storage/internal/settings/DataType.java:28]: Why is the unsorted implementation also created by calling `mapProvider.getSortedMapImplementation()`. You could also just do `new HashMap<>()` if I am not mistaken.

File: CodeStyle.xml [hunk 7]
```diff
@@ -1,885 +1,885 @@
...
+    </codeStyleSettings>
+    <codeStyleSettings language="protobuf">
+        <option name="RIGHT_MARGIN" value="80"/>
+        <indentOptions>
+            <option name="INDENT_SIZE" value="2"/>
+            <option name="CONTINUATION_INDENT_SIZE" value="2"/>
+            <option name="TAB_SIZE" value="2"/>
+        </indentOptions>
+    </codeStyleSettings>
 </code_scheme>
```
Comment (KotlinFactory) [inline CodeStyle.xml:885]: Why was this changed? It should stay this way.

**Comment by KotlinFactory** [comment]
Thank you so much for your dedication and work. I think its great that you want to improve SimplixStorage. Please split your PR into smaller pieces (One for each point (Like remove redundant toString() call)) and we can approve them.

Regards,
Leonhard

File: .github/ISSUE_TEMPLATE/bug_report.md (hunk 1 above)
Comment (ProfliX) [reply .github/ISSUE_TEMPLATE/bug_report.md:2]: Yep..
